package.json

```
@@ -52,7 +51,11 @@
 		"terser": "5.31.0"
 	},
 	"devDependencies": {
-		"@playform/build": "0.0.10"
+		"@playform/build": "0.0.10",
+		"astro": "4.10.0"
+	},
+	"peerDependencies": {
+		"astro": "^4.10.0"
```

How about turning this into a wildcard to allow all possible Astro transformer definitions?

```suggestion
		"astro": "*"
```

---

I think it's safer to support only minor and patch versions?

---

Will adding `astro` to `devDependencies` and `peerDependencies` solve for future versions of `astro` as well ?

For example which version will `npm` select during a normal `npm install` if there's an `astro` 5.10 and in `peerDependencies` we have `astro` `^4.10` but for the `devDependencies` we still use `4.10`.

Also I think this - https://github.com/PlayForm/Compress/pull/370/files#r1630294340 is a good suggestion, as it may match the `astro` version correctly and allow for backwards compatibility, however, still unsure. That's why we have `@playform/adapt` - https://github.com/PlayForm/Adapt in development but it's still early stages.

---

Yes.
It supports Astro v4.x.x

I don't think it's a good idea to support Astro 5, as there may be changes there that will change the plugins API.

The best solution would be to leave it as is. With caret.

---

What I'm wondering is if we leave it as is:

```json
{
	"devDependencies": {
		"@playform/build": "0.0.10",
		"astro": "4.10.0"
	},
	"peerDependencies": {
		"astro": "^4.10.0"
	}
}
```

and a new version comes out, say `4.20` or `4.30`. Will that break because we still use `astro` `4.10.0` in the `devDependencies` and which `peerDependencies` versions will get selected and installed alongside `astro-compress`.

Can you test this with a lower version?

For example have a `package.json` with:

```json
{
	"devDependencies": {
		"@playform/build": "0.0.10",
		"astro": "4.3.5"
	},
	"peerDependencies": {
		"astro": "^4.3.5"
	}
}
```

and test it with an `astro` installation of `4.10.1`. This is what I'm wondering.

---

So I've issued new versions of `astro-compress` [v2.2.27](https://github.com/PlayForm/Compress/releases/tag/AstroCompress%2Fv2.2.27) and `@playform/compress` [v0.0.12](https://github.com/PlayForm/Compress/releases/tag/Compress%2Fv0.0.12) that should address this using the `astro` `*` wildcard version. Can you give it a go ? I've tested locally and it seems to install the correct version used in the parent `astro` installation.